farmhelden/rest_server/serializers.py

```diff
@@ -161,19 +161,22 @@ def validate(self, data):
         }
 
 
-class CampaignSerializer(serializers.HyperlinkedModelSerializer):
+class LocationSerializer(serializers.HyperlinkedModelSerializer):
     class Meta:
-        model = Campaign
-        fields = ['id', 'farm_id', 'location_id', 'date_from', 'date_to']
+        model = Location
+        fields = ['id', 'point', 'info', 'farm_id', 'location_type']
 
     def create(self, validated_data):
-        return Campaign.objects.create(**validated_data)
+        return Location.objects.create(**validated_data)
 
 
-class LocationSerializer(serializers.HyperlinkedModelSerializer):
+class CampaignSerializer(serializers.HyperlinkedModelSerializer):
+    location = LocationSerializer(read_only=True, source="location_id")
+
     class Meta:
-        model = Location
-        fields = ['id', 'point', 'info', 'farm_id', 'location_type']
+        model = Campaign
+        fields = ['id', 'farm_id', 'location', 'date_from', 'date_to']
```

Comment:
```suggestion
        fields = ['id', 'farm_id', 'location', 'location_id', 'date_from', 'date_to']
```
Ich denke beide Felder sind hier hilfreich, so muss man nicht erst in die Location um nach der ID zu schauen

Reply:
die location_id ist in der response schon drin:

    "results": [
        {
            "id": 1,
            "farm_id": "http://localhost:8000/farms/4/",
            "location": {
                "id": 2,
                "point": null,
                "info": "Ein schöner ort",
                "farm_id": "http://localhost:8000/farms/4/",
                "location_type": "1"
            },
            "date_from": "2020-05-01T00:00:00Z",
            "date_to": "2020-05-05T00:00:00Z"
        }
]

Reply:
Jo genau, mit der Änderung wäre das dann ja:
```
"results": [
    {
        "id": 1,
        "farm_id": "http://localhost:8000/farms/4/",
        "location_id": 2,
        "location": {
            "id": 2,
            "point": null,
            "info": "Ein schöner ort",
            "farm_id": "http://localhost:8000/farms/4/",
            "location_type": "1"
        },
        "date_from": "2020-05-01T00:00:00Z",
        "date_to": "2020-05-05T00:00:00Z"
    }
```

Reply:
So könnte man im Frontend einfacher nach location_id filtern und muss nicht erst schauen ob location gesetzt ist und dann location.id aufrufen.